### ISSUE TITLE
Bug when building TPC-C and TPC-H on Azure

### ISSUE THREAD

**Issue by jamiere-msft** [TPC-Council/HammerDB#58]
When trying to build a database in Azure using the GUI, there is an inconsistency in the database variable.  Azure DB does not support the USE <dbname> syntax and the GUI supports this.  In the generated load scripts there is a bug that causes the USE syntax to be used.
In the generated load script, with the Azure check box checked, there are the following lines...
17: set azure "true";#Azure Type Connection
18: set database "tpch";# Database containing the TPC Schema
Later there is this...
41: if { $azure eq "true" } { append connection ";" "DATABASE=$db" }
The variable set in line 18 (database) does not match what is use in line 41 (db).  This causes subsequent lines, like lines 187-188 to issue the USE command.  

The fix is to change the variable call in line 41 to be "$database", not "$db".

**Comment by sm-shaw** [comment]
I currently don't have access to an azure database for debugging. However if I follow the issue correctly once azure has been specified then "USE database_name" should not be used as it causes an error and is not ignored.  This occurs twice in the TPCH workload - once for the query sets and once for the refresh and once each in the test and timed TPC-C workloads. If this is correct then it is a case of modifying the use line as follows: 

`if {!$azure} {odbc "use $database"}`

This would be like this for the 4 cases found.  Please confirm if this is the correct functionality needed for azure. 
```
TPCH
#TPCH QUERY SETS PROCEDURE
database connect odbc $connection
if {!$azure} {odbc "use $db"}
odbc set autocommit off

#REFRESH FUNCTION
database connect odbc $connection
if {!$azure} {odbc "use $database"}
odbc set autocommit off

TPCC
TIMED
#RUN TPC-C
database connect odbc $connection
if {!$azure} {odbc "use $database"}
odbc set autocommit off

TEST
#RUN TPC-C
database connect odbc $connection
if {!$azure} {odbc "use $database"}
odbc set autocommit off
}
```

**Comment by sm-shaw** [comment]
Also what is needed for the TPC-C and TPC-H builds. There are a number of uses of "USE" in here.  Do these also need updating for azure or is it just some of them? ie on Azure it will find a database that exists so it is only when the worker/builder threads are logging on that "USE" needs adapting? 

TPC-C BUILD
```
proc CreateDatabase { odbc db imdb } {
set table_count 0
puts "CHECKING IF DATABASE $db EXISTS"
set db_exists [ odbc "IF DB_ID('$db') is not null SELECT 1 AS res ELSE SELECT 0 AS res" ]
if { $db_exists } {
odbc "use $db"

CreateDatabase odbc $db $imdb 
odbc "use $db"

odbc "use $db"
odbc set autocommit off 

set sql(1) "USE $db"
set sql(2) "EXEC sp_updatestats"
```
TPC-H BUILD

```
puts "UPDATING SCHEMA STATISTICS"
set sql(1) "USE $db"
set sql(2) "EXEC sp_updatestats"

et db_exists [ odbc "IF DB_ID('$db') is not null SELECT 1 AS res ELSE SELECT 0 AS res" ]
if { $db_exists } {
odbc "use $db"

CreateDatabase odbc $db
odbc "use $db"

odbc "use $db"
odbc set autocommit off 
```

**Comment by jamiere-msft** [comment]
Hi Steve,

The issue impacts both C and H, build and runtime.  It is also a little more nuanced than I originally stated.  If you are using SQL Server on an Azure VM then the USE command is appropriate.  The difference pops up when using SQL DB.  That is the managed instance where when you connect to the system you are automatically “attached” to the database and the USE command generates an error.

For our SQL DB testing we were using a backdoor and restoring a database on the service.  I need to run through a build pointing at SQL DB and see what happens.  We know the runtime generates an error.

Jamie Reding | Microsoft SQL Server | 425-703-0510

**Comment by sm-shaw** [comment]
Fixed in v3.2
